packages/SystemsThinkingWorkbench-Core.package/STWNode.class/instance/updateFilling.st

```diff
@@ -0,0 +1,13 @@
+geometry
+updateFilling
+	"Calculates size of the circle representing filling state by using a logistic function.
+	If the level is below 0, the circle is hidden.
+	The size is set and the circle is centered in the node afterwards."
+	
+	| diameter |	
+
+	self level <= 0 ifTrue: [self fillingMorph hide]
+					ifFalse: [diameter := (1 / (1 + (STWNode fillingFunctionSteepness negated * (self level - STWNode fillingFunctionMiddle)) exp)) sqrt * (self width - 1).
```

Comment:
I am going to leave a note here. We should talk about the formatting of this function in the next XCOM.